### PR TITLE
Generate trim-safe module metadata

### LINKED ISSUE #3226: Generate module discovery and DI registration metadata for trim safety
Parent: #2996

## Scope

Eliminate runtime assembly/type scanning from the normal module-discovery and dependency-injection path.

- Extend `ModularPipelines.SourceGenerator` to emit explicit metadata/registration for statically discoverable modules and their dependency types.
- Make `ModuleAutoRegistrar`, `AssemblyLoadedTypesProvider`, and related `ServiceCollectionExtensions` consume generated metadata first.
- Preserve a clearly annotated reflection fallback for dynamic/plugin assemblies that cannot be known at compile time.
- Preserve auto-registration, dependency resolution, and unused-module detection behavior.
- Add generator and runtime regression tests covering inheritance, dependency-only modules, duplicate registration, and the dynamic fallback.
- Build core with trim analysis enabled and introduce no new IL warnings.

This is the first remaining Native-AOT blocker after #3100 generated command/secret metadata.

### PR DESCRIPTION
**Pull request by thomhurst**
## Summary
- emit assembly module discovery, DI registration, and direct dependency metadata
- prefer generated metadata in registration, auto-registration, dependency resolution, and unused-module discovery
- retain annotated reflection fallback for dynamic or incomplete assemblies

## Validation
- core Release build (0 errors; existing warnings only)
- source-generator tests: 17 passed
- generated metadata runtime tests: 4 passed
- dependency/cycle/unused/loaded-type regressions: 27 passed
- targeted whitespace verification

Closes #3226

### CODE REVIEW

**Comment by thomhurst** [comment]
@codex review

**Comment by thomhurst** [comment]
@codex review

**Comment by thomhurst** [comment]
Fixed the analyzer gate in 4600ac8d31. The legacy runtime-Type activator is now explicitly documented/suppressed as the reflection fallback; generated registrations continue through the annotated generic activator. Validation: CI-equivalent changed-file analyzer gate passed with EnableCiAnalyzers=true; core Release build passed (0 errors; existing warnings); whitespace gate; git diff --check. @codex review

**Comment by thomhurst** [comment]
Addressed the latest review cleanup in `4d5a05be21`:

- centralized partial-load-safe `Assembly.GetTypes()` fallback in `AssemblyTypeLoader`
- centralized generic/runtime activation context handling in `ModuleActivator`
- documented why generated metadata remains incomplete and linked the follow-up to #3228

Validation:
- `GeneratedModuleMetadataTests`: 4/4
- `ModuleMetadataGeneratorTests`: 10/10
- `ModuleConfigureTests`: 6/6
- changed-file `dotnet format` info + whitespace gates
- `ModularPipelines.sln` Release build with `EnableCiAnalyzers=true`: 0 errors (existing warnings only)

@codex review

**Comment by thomhurst** [comment]
@codex review

**Comment by thomhurst** [comment]
@codex review

File: src/ModularPipelines.SourceGenerator/ModuleMetadataGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+            || type.IsAbstract
+            || type.IsGenericType
+            || !ImplementsModule(type, context.SemanticModel.Compilation))
+        {
+            return null;
+        }
+
+        var currentAssembly = context.SemanticModel.Compilation.Assembly;
+        var dependencies = ImmutableArray.CreateBuilder<DependencyMetadataInfo>();
+        var dependenciesComplete = true;
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/ModularPipelines.SourceGenerator/ModuleMetadataGenerator.cs]: Fixed in c770eba304. Partial modules now emit dependenciesComplete: false, preserving reflection fallback for attributes another generator may add; non-partial modules retain complete metadata.

**Comment by thomhurst** [comment]
@codex review
